### PR TITLE
Closes #56

### LINKED ISSUE #56: `Model` class clean up
To do:

- [x] Revise documentation
  - [x] RStudio outline section dividers (use `# function or class name ----`)
  - [x] Add lifecycle badge (as a default take "stable" for existing functions or classes)
  - [x] Replace `##'` by `#'`
  - [x] Add to `pkgdown.yml`
  - [x] Remove file history and the timestamp in comments on top of file
- [x] Add unit tests
  - [x] For classes:
    - [x] Test default constructor
    - [x] For reference classes: test methods
  - [x] For functions: as usual
- [x] Refactor as needed
  - [x] Code styling (indentation etc.) - install package `styler` and use corresponding add-in
  - [x] Keep externally used names so that we don't break downstream code
  - [x] Remove `keywords` tags
  - [x] For functions or methods:
    - [x] Add assertions for arguments using `checkmate` functions
    - [x] If it is too long, break up into pieces (rule of thumb: maximum 50 lines of code)
    - [x] Consider renaming internal variables etc. if needed
    - [x] Avoid explicit use of `return`
  - [x] For classes:
    - [x] move function definitions outside if possible (e.g. move validation function as `validity_subjects()` to `Data-validity.R` and have corresponding test file `test-Data-validity.R`)
    - [x] In `setClass` function, just use the `slots` instead of outdated `representation`
    - [x] Remove things like `validObject(.Data())`

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #56

### CODE REVIEW

**Review by danielinteractive — COMMENTED** [review]
thx Wojciech, pls see inside comment on making argument check more precise

File: R/Model-validity.R [hunk 1]
```diff
@@ -21,3 +21,20 @@ validate_general_model <- function(object) {
...
 }
+
+#' @describeIn validate_model_objects validates that the names of the
+#'   arguments in `dose` and `prob` functions contains `prob` and `dose`
+#'   respectively, as well as they match `sample` slot of the `object`.
+validate_model <- function(object) {
+  o <- Validate()
+
+  o$check(
+    all(names(formals(object@dose)) %in% c("prob", object@sample)),
```
[the remark this answers is not in the document]
Comment (wwojciech) [reply R/Model-validity.R]: Thank you for this suggestion @danielinteractive I now proposed something which I hope seems adequate to the (current and future) needs.

**Comment by wwojciech** [comment]
@dinakar29 , @danielinteractive - do you have any idea why the validation fails here? I think it has something to do with staged dependencies? Thanks for helping me with this!

**Comment by dinakar29** [comment]
There's an ongoing issue with public RSPM. We'll need to wait till RStudio fixes this and rerun the pipelines.
image

**Review by danielinteractive — APPROVED** [review]
very nice helper, thx Wojciech!

**Comment by wwojciech** [comment]
> very nice helper, thx Wojciech!

Thank you Daniel (@danielinteractive)!

**Pull request merged**